six/modules/c++/six.sicd/source/ComplexData.cpp

```diff
@@ -56,38 +56,42 @@ void ComplexData::getOutputPlaneOffsetAndExtent(
         types::RowCol<size_t>& offset,
         types::RowCol<size_t>& extent) const
 {
-    offset.row = offset.col = 0;
-
     const six::sicd::AreaPlane& areaPlane =
         *radarCollection->area->plane;
+    getOutputPlaneOffsetAndExtent(offset, extent, areaPlane);
+}
+
+void ComplexData::getOutputPlaneOffsetAndExtent(
+        types::RowCol<size_t>& offset,
+        types::RowCol<size_t>& extent,
+        const AreaPlane& areaPlane) const
+{
+    offset.row = offset.col = 0;
     extent.row = areaPlane.xDirection->elements;
     extent.col = areaPlane.yDirection->elements;
 
     if (imageFormation.get())
     {
         const std::string& segmentID =
                 imageFormation->segmentIdentifier;
+        try
+        {
+            const six::sicd::Segment& segment = areaPlane.getSegment(segmentID);
```

Comment:
I'm lost on how the logic should be here. So if segmentID is empty, nothing should happen. If there is a segmentID, but it doesn't match anything in the list, it should throw. Is that correct?

I'm concerned that with an extra `hasSegment()` method, then to get a segment you'd have to loop over the list once to see if it's there, then again to actually get it. I'm not sure how long they normally are. What about replacing `getSegment()` with `getSegmentIndex()`?